drivers/wifi/eswifi/eswifi_offload.c

```diff
@@ -0,0 +1,510 @@
+			     eswifi->buf, sizeof(eswifi->buf));
+	if (err || !__is_at_ok(eswifi->buf)) {
+		SYS_LOG_ERR("Unable to set remote port");
+		eswifi_unlock(eswifi);
+		return -EIO;
+	}
+
+	/* Start TCP client */
+	sprintf(eswifi->buf, "P6=1\r");
+	err = eswifi_request(eswifi, eswifi->buf, strlen(eswifi->buf),
+			     eswifi->buf, sizeof(eswifi->buf));
+	if (err || !__is_at_ok(eswifi->buf)) {
+		SYS_LOG_ERR("Unable to connect");
+		eswifi_unlock(eswifi);
+		return -EIO;
+	}
+	SYS_LOG_ERR("%s", eswifi->buf);
+
+	socket->state = ESWIFI_SOCKET_STATE_CONNECTED;
+
+	eswifi_unlock(eswifi);
+
+	return 0;
+}
+
+static int eswifi_off_accept(struct net_context *context,
+		     net_tcp_accept_cb_t cb,
+		     s32_t timeout,
+		     void *user_data)
+{
+	/* TODO */
+	SYS_LOG_DBG("");
+	return -ENOTSUP;
+}
+
+static int eswifi_off_send(struct net_pkt *pkt,
+			   net_context_send_cb_t cb,
+			   s32_t timeout,
+			   void *token,
+			   void *user_data)
+{
+	struct eswifi_off_socket *socket = pkt->context->offload_context;
+	struct eswifi_dev *eswifi = eswifi_by_iface_idx(socket->context->iface);
+	unsigned int bytes;
+	struct net_buf *frag;
+	int err, offset;
+
+	bytes = net_pkt_get_len(pkt);
+
+	SYS_LOG_DBG("%u bytes", bytes);
+
+	eswifi_lock(eswifi);
+
+	__select_socket(eswifi, socket->index);
+
+	/* header */
+	sprintf(eswifi->buf, "S3=%u\r", bytes);
+
+	/* copy payload */
+	offset = strlen(eswifi->buf);
+	for (frag = pkt->frags; frag; frag = frag->frags) {
+		memcpy(&eswifi->buf[offset], frag->data, frag->len);
+		offset += frag->len;
+	}
+
+	err = eswifi_request(eswifi, eswifi->buf, offset + 1,
+			     eswifi->buf, sizeof(eswifi->buf));
+	if (err || !__is_at_ok(eswifi->buf)) {
+		SYS_LOG_ERR("Unable to send data");
+		eswifi_unlock(eswifi);
+		return -EIO;
+	}
+
+	net_pkt_unref(pkt);
+	eswifi_unlock(eswifi);
+
+	return 0;
+}
+
+static int eswifi_off_sendto(struct net_pkt *pkt,
+			     const struct sockaddr *dst_addr,
+			     socklen_t addrlen,
+			     net_context_send_cb_t cb,
+			     s32_t timeout,
+			     void *token,
+			     void *user_data)
+{
+	/* TODO */
+	SYS_LOG_DBG("");
+	return -ENOTSUP;
+}
+
+static int eswifi_off_recv(struct net_context *context,
+			   net_context_recv_cb_t cb,
+			   s32_t timeout,
+			   void *user_data)
+{
+	struct eswifi_off_socket *socket = context->offload_context;
+	struct eswifi_dev *eswifi = eswifi_by_iface_idx(context->iface);
+
+	/* TODO */
+	SYS_LOG_DBG("");
+
+	eswifi_lock(eswifi);
+	socket->recv_cb = cb;
+	socket->user_data = user_data;
+	eswifi_unlock(eswifi);
+
+	return 0;
+}
+
+static int eswifi_off_put(struct net_context *context)
+{
+	struct eswifi_off_socket *socket = context->offload_context;
+	struct eswifi_dev *eswifi = eswifi_by_iface_idx(context->iface);
+	int err;
+
+	SYS_LOG_DBG("");
+
+	eswifi_lock(eswifi);
+
+	__select_socket(eswifi, socket->index);
+
+	socket->context = NULL;
+	socket->state = ESWIFI_SOCKET_STATE_NONE;
+
+	/* Stop TCP client */
+	sprintf(eswifi->buf, "P6=0\r");
```

Comment:
btw, you should always use snprintf() or better snprintkf() so that there is no change to overflow the target buffer.